### PR TITLE
First version for options in a dict

### PR DESCRIPTION
**Pull request by rokdd**
First version for a dict handling the options. No error handling if the wrong type is given. Planned a structure like:

```
WAGTAILMARKDOWN = { 
#this I would not name with bleach.. 
'allowed_tags': [],
...
extensions; {},
extensions_config: {},
..
}
```

### CODE REVIEW

**Comment by zerolab** [comment]
Hey @rokondo, thanks for this.
It is along the lines I was thinking of, so you're up to a great start

To finish off, this PR needs:
- updating the docs (well, the README)
- to be a bit more tidy up (e.g. `allowed_styles`, `allowed_tags` are lists which when added may result in duplicates. probably ok for bleach, but a nice thing to have)
- cater for the `WAGTAILMARKDOWN_AUTODOWNLOAD_FONTAWESOME` setting (https://github.com/torchbox/wagtail-markdown/blob/main/wagtailmarkdown/widgets.py#L22)
- and maybe raising deprecation warnings for the separate settings so that people can update to the new format.

**Comment by rokondo** [comment]
Well nope,will do it tomorrow. With the last one I thought to still accept the old prefs. But obviously you agree to my concept of the options.

File: wagtailmarkdown/utils.py [hunk 1]
```diff
@@ -154,6 +155,10 @@ def _get_markdown_kwargs():
 
     if hasattr(settings, "WAGTAILMARKDOWN_EXTENSIONS"):
         markdown_kwargs["extensions"] += settings.WAGTAILMARKDOWN_EXTENSIONS
+        warnings.warn(
+            "WAGTAILMARKDOWN_EXTENSIONS will be deprecated in version 7.1, use WAGTAILMARKDOWN = { extensions: {} } as dict instead",
```
Comment (zerolab) [inline wagtailmarkdown/utils.py:159]:
```suggestion
            "WAGTAILMARKDOWN_EXTENSIONS will be deprecated in version 0.9.0, use WAGTAILMARKDOWN = { extensions: {} } as dict instead",
```

My plan is to get this in v0.7.0, so giving it one extra release for the deprecations to be addressed

Comment (rokondo) [reply wagtailmarkdown/utils.py:159]: I will do..

**Comment by rokondo** [comment]
I changed the readme also a bit by structure.. I am finished from my side.

File: README.md [hunk 2]
```diff
@@ -29,27 +29,25 @@ extensions to make it actually useful in Wagtail:
...
-```python
-WAGTAILMARKDOWN_EXTENSIONS = ["toc", "sane_lists"]
-```
 
-Extensions can be configured too:
+### Configuration
 
-```python
-WAGTAILMARKDOWN_EXTENSIONs_CONFIG = {'pymdownx.arithmatex': {'generic': True}}
+With version 0.7.0 settings are defined as dict (The previous settings with constants will be deprecated at version 0.9.0):
```
Comment (tomdyson) [inline README.md:38]:
```suggestion
In version 0.7.0, settings are defined as a dict (the previous settings with constants will be deprecated in version 0.9.0):
```

File: wagtailmarkdown/utils.py [hunk 3]
```diff
@@ -146,6 +155,12 @@ def _get_markdown_kwargs():
 
     if hasattr(settings, "WAGTAILMARKDOWN_EXTENSIONS"):
         markdown_kwargs["extensions"] += settings.WAGTAILMARKDOWN_EXTENSIONS
+        warnings.warn(
+            "WAGTAILMARKDOWN_EXTENSIONS will be deprecated in version  0.9.0, use WAGTAILMARKDOWN = { extensions: {} } as dict instead",
```
Comment (tomdyson) [inline wagtailmarkdown/utils.py:159]:
```suggestion
            "WAGTAILMARKDOWN_EXTENSIONS will be deprecated in version 0.9.0. Use WAGTAILMARKDOWN = { extensions: {} } as a dict instead",
```

**Review by tomdyson — COMMENTED** [review]
Two tiny tweaks. Elsewhere, the quoting style is inconsistent in Python code. Can we use `black` from now on?

**Comment by rokondo** [comment]
> 
> 
> Two tiny tweaks. Elsewhere, the quoting style is inconsistent in Python code. Can we use `black` from now on?

Thanks.. shall I change to `black` ?

**Comment by zerolab** [comment]
We have [pre-commit](https://pre-commit.com/) support - https://github.com/torchbox/wagtail-markdown/blob/main/.pre-commit-config.yaml

best to install `pre-commit` locally:
```sh
$ pip install pre-commit
# in project dir
$ pre-commit install
# run all checks once for this, then the checks will run only on the changed files
$ pre-commit run --all-files
```

**Comment by rokondo** [comment]
Thanks for the pre-commit tool - pretty cool. Hope it is fine now.

**Comment by zerolab** [comment]
Thank you for this PR @rokdd 
Tidied it up and merged in 99f6cd6 

While doing the tidy up, I realised `WAGTAILMARKDOWN_EXTENSIONS`, `WAGTAILMARKDOWN_EXTENSIONS_CONFIG` and `WAGTAILMARKDOWN_AUTODOWNLOAD_FONTAWESOME` were all in a pre-release so decided it is cleaner to do away with the deprecation warnings. Sorry for the extra work!

**Pull request closed without merging**